EasyPost/Hooks.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Net.Http;
+
+namespace EasyPost;
+public class Hooks
+{
+    /// <summary>
+    ///     An <see cref="EventHandler{RequestBeforeExecutionEventArgs}"/> to view an HTTP request by the client prior to being sent.
+    ///     Editing the <see cref="HttpRequestMessage"/> in this callback does not impact the <see cref="HttpRequestMessage"/> being executed.
+    /// </summary>
+    public EventHandler<RequestBeforeExecutionEventArgs>? OnRequestBeforeExecution { get; set; }
+
+    /// <summary>
+    ///     An <see cref="EventHandler{RequestResponseReceivedEventArgs}"/> to view an HTTP response received by the client.
+    /// </summary>
+    public EventHandler<RequestResponseReceivedEventArgs>? OnRequestResponseReceived { get; set; }
+}
+
+public class RequestBeforeExecutionEventArgs : EventArgs
+{
+    /// <summary>
+    ///     The <see cref="HttpRequestMessage"/> about to be executed by the HTTP request.
+    /// </summary>
+    public HttpRequestMessage Request { get; }
+
+    /// <summary>
+    ///     The timestamp of the HTTP request.
+    /// </summary>
+    public int Timestamp { get; }
+
+    /// <summary>
+    ///     A unique identifier for the HTTP request-response pair.
+    /// </summary>
+    public Guid Id { get; }
+
+    /// <summary>
+    ///     Constructs a new instance of the <see cref="RequestBeforeExecutionEventArgs"/> class.
+    /// </summary>
+    /// <param name="request">The <see cref="HttpRequestMessage"/> about to be executed by the HTTP request.</param>
+    /// <param name="timestamp">The timestamp of the HTTP request.</param>
+    /// <param name="guid">A unique identifier for the HTTP request-response pair.</param>
+    internal RequestBeforeExecutionEventArgs(HttpRequestMessage request, int timestamp, Guid guid)
+    {
+        Request = request;
+        Timestamp = timestamp;
+        Id = guid;
+    }
+}
+
+public class RequestResponseReceivedEventArgs : EventArgs
```

Comment:
issue: missing `method`, `path`, `headers`. If they are in the response param, we need to break them out

Reply:
Resolved, following spec.